src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -489,6 +525,16 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
             )
         end
 
+        # Environment precipitation sources (to be applied to grid mean)
+        @. S_q_tot⁰_prev_level = q_tot_precipitation_sources(
+            precip_model,
+            thermo_params,
+            microphys_params,
+            dt,
+            q_tot_prev_level,
+            ts_prev_level,
+        )
+
```

Comment:
Since we don't need this in the integral, we can actually move it outside the loop and calculate the environment precipitation of the entire 3d field.

Reply:
Sounds good. Will update tomorrow